### ISSUE TITLE
[BUG] Proxy environment variables ignored

### ISSUE THREAD

**Issue by lscorcia** [npm/cli#6930]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

Hi, in our CI/CD environment we am using the latest LTS node (20.9.0) which includes npm 10.1.0. Builds are executed in a linux docker container. We have a company http proxy that allows outgoing requests - both the docker host and container have the http_proxy and https_proxy environment variables correctly set. Everything was working fine with the previous LTS (v18) and from my tests it was working fine with v19 too. Since v20 became the new LTS however it seems the environment variables are not picked up, leading to errors like this:

```
root@a1420f619e6f:/# npm install -g https://tls-test.npmjs.com/tls-test-1.0.0.tgz
npm ERR! code ENOTFOUND
npm ERR! syscall getaddrinfo
npm ERR! errno ENOTFOUND
npm ERR! network request to https://tls-test.npmjs.com/tls-test-1.0.0.tgz failed, reason: getaddrinfo ENOTFOUND tls-test.npmjs.com
npm ERR! network This is a problem related to network connectivity.
npm ERR! network In most cases you are behind a proxy or have bad network settings.
npm ERR! network
npm ERR! network If you are behind a proxy, please make sure that the
npm ERR! network 'proxy' config is set properly.  See: 'npm help config'

npm ERR! A complete log of this run can be found in: /root/.npm/_logs/2023-10-24T18_14_22_344Z-debug-0.log
```

The error disappears if I explicitly set the proxy for npm with `npm config set proxy http://proxyhost:8080` but this workaround is not acceptable in our environment. We need it to pick up the environment configs.

### Expected Behavior

npm should use the http_proxy and https_proxy environment variables to access the registry if they are set.

### Steps To Reproduce

On a docker host with an http proxy, run the following commands:

1. `docker run --rm -it node:20 /bin/bash`
2. Check that http_proxy/https_proxy have been carried over to the container by running the env command:
```
root@be39f202b0ce:/# env
HOSTNAME=be39f202b0ce
YARN_VERSION=1.22.19
PWD=/
HOME=/root
https_proxy=http://proxyhost:8080/
TERM=xterm
SHLVL=1
HTTPS_PROXY=http://proxyhost:8080/
HTTP_PROXY=http://proxyhost:8080/
http_proxy=http://proxyhost:8080/
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
NODE_VERSION=20.8.1
_=/usr/bin/env

```
4. `npm install -g https://tls-test.npmjs.com/tls-test-1.0.0.tgz`
5. See error. The proxy environment vars are ignored and access to the registry fails.

### Environment

- npm: 10.1.0
- Node.js: 20.8.1
- OS Name: Official node 20 docker image
- npm config:
```ini
; node bin location = /usr/local/bin/node
; node version = v20.8.1
; npm local prefix = /
; npm version = 10.1.0
; cwd = /
; HOME = /root
; Run `npm config ls -l` to show all defaults.
```

**Comment by lscorcia** [comment]
Sorry, duplicate of #6835 . Also fixed in 10.2.0. Shouldn't the fix be backported to the latest LTS?

**Comment by tonyvargese** [comment]
I have similar issue
